torsiondrive/dihedral_scanner.py

```diff
@@ -455,6 +468,7 @@ def launch_constrained_opt(self, molecule, grid_id):
             print("Launching new job at %s" % new_job_path)
         # launch optimization job inside scratch folder
         self.engine.M = copy.deepcopy(molecule)
+        self.engine.extra_constraints = self.extra_constraints
```

Comment:
Note: extra_constraints is passed to `engine` here to allow the use of engine to be consistent with current scan. The idea is that multiple DihedralScanner instances could potentially use the same `Engine` instance for running running constrained optimizations.